docs/docs/database-seeds.md

```diff
@@ -0,0 +1,223 @@
+
+* An admin user so that you can log in to your new instance
+* A list of categories that can be assigned to a Product
+* Lists of roles and permissions
+
+Seed data is not meant for:
+
+* Sample data to be used in development
+* Data to run tests against
+* Randomized data
+
+## Best Practices
+
+Ideally seed data should be idempotent: you can execute the seed
+script against your database at any time and end up with the seed data properly
+populated in the database. It should not result in wiping out existing records
+or creating duplicates of any seeded data that is already present.
+
+Making your seeds idempotent requires more code than just a straight
+`createMany()`. The code examples below use the safest idempotent strategy
+by having an `upsert` check if a record with the same unique identifier
+already exists, and if so just update it, if not then create it. But, this
+technique requires a separate SQL statement for each member of your data array
+and is less performant than `createMany()`.
+
+You could also do a check if *any* data exists in the database first, and if
+not, create the records with `createMany()`. However, this means that any
+existing seed data that may have been modified will remain, and would not be
+updated to match what you expect in your seed.
+
+When in doubt, `upsert`!
+
+## When seeds run
+
+Seeds are automatically run the first time you migrate your database:
+
+```bash
+yarn rw prisma migrate dev
+```
+
+They are run *every* time you reset your database:
+
+```bash
+yarn rw prisma migrate reset
+```
+
+You can manually run seeds at any time with the following command:
+
+```
+yarn rw prisma db seed
+```
+
+You generally don't need to keep invoking your seeds over and over again, so it
+makes sense that Prisma only does it on a complete database reset, or when the
+database is created with the first `prisma migrate dev` execution. But as your
+schema evolves you may add a new model that requires some seeded data and so
+you can add it to your seed file and then manually run it to create those
+records.
+
+### Performance
+
+Prisma is faster at execting a `createMany()` instead of many `create` or
+`upsert` functions. Unfortunately, you lose the ability to easily make your seed
+idempotent with a single function call.
+
+One solution to simulate an `upsert` will still using `createMany()` could be
+to start with the full array of data and first check to see whether each of
+those records already exist in the database. If they do, create two
+arrays: one for records that don't exist and run `createMany()` with them, and
+the second list for records that do exist, and run `updateMany()` on those.
+
+Unfortunately this relies on a select query for each record, which may negate
+the performance benefits of `createMany()`. Since you are running seeds
+realitively rarely, it's our recommendation that you focus less on absolute
+performance and worry more about making them easy to maintain.
```

Comment:
I feel like this entire #### Performance section could just have been this 😀 You run seeds like once. Who cares about perf 😆

Reply:
haha I agree, although DT does make a good point with `createMany()`: in my later docs example with the book categories, there are apparently 3,000+ official ones! If you're doing a lot of CI that's one that could probably stand to be as fast as possible, and really no reason to do an `upsert()` since you won't be manually editing the list and need to restore changed records!